lib/cartopy/mpl/gridliner.py

```diff
@@ -362,22 +434,88 @@ def _draw_gridliner(self, nx=None, ny=None, background_patch=None):
+            # if it's default, then modify; if not, don't touch
+            if self.xlocator == degree_locator:
+                # reduce cramping, especially for those projections
+                if (isinstance(proj, _CRAMPED_PROJS) or
+                        np.diff(x_label_points).mean() <= 20):
+                    x_label_points = x_label_points[1::2]
+
+        if self.y_inline or self.x_inline:
+            if self.ylocator == degree_locator:
+                if (isinstance(proj, _CRAMPED_PROJS) or
+                        np.diff(y_label_points).mean() <= 20):
+                    y_label_points = y_label_points[::2]
+
+        # prevent overlap with existing x_inline coordinates
+        existing_coords = []
+
         if self.xlabels_bottom or self.xlabels_top:
-            self._assert_can_draw_ticks()
             self.xformatter.set_locs(x_label_points)
             for x in x_label_points:
+                if self.x_inline:
+                    other_value = np.median(y_label_points)
+                    coordinate = (x, other_value)
+                    if coordinate in existing_coords:
+                        continue
+                    elif (isinstance(proj, cartopy.crs.TransverseMercator) and
+                          x in (-120, 120)):
+                        continue
+                    else:
+                        existing_coords.append(coordinate)
+                        # remove overlaps
+                        if x == -180 or x == 180 or x == 0:
+                            existing_coords.append((-x, other_value))
+                            existing_coords.append((0, other_value))
+                    clip_on = True
+                elif self.y_inline:
+                    other_value = np.nanpercentile(y_ticks, 0.4)
+                    clip_on = False
+                else:
+                    if (isinstance(proj, cartopy.crs.RotatedPole) and
+                            x == -180):
+                        continue
+                    other_value = None
+                    clip_on = False
+
                 if self.xlabels_bottom:
-                    self._add_gridline_label(x, axis='x', upper_end=False)
-                if self.xlabels_top:
+                    self._add_gridline_label(x, axis='x', upper_end=False,
+                                             other_value=other_value,
+                                             clip_on=clip_on)
+                if self.xlabels_top and not (self.x_inline or self.y_inline):
                     self._add_gridline_label(x, axis='x', upper_end=True)
 
         if self.ylabels_left or self.ylabels_right:
-            self._assert_can_draw_ticks()
             self.yformatter.set_locs(y_label_points)
             for y in y_label_points:
+                if self.y_inline:
+                    if y == 0:  # (0, 0) already plotted from x
+                        continue
+                    if isinstance(proj, cartopy.crs.RotatedPole):
```

Comment:
Have you tested this with various choices of parameters to this projection?

Reply:
Done see notebook linked.